models/running_output.go

```diff
@@ -246,7 +246,7 @@ func (r *RunningOutput) write(metrics []telegraf.Metric) error {
 	r.WriteTime.Incr(elapsed.Nanoseconds())
 
 	if err == nil {
-		r.log.Debugf("Wrote batch of %d metrics in %s", len(metrics), elapsed)
+		r.log.Debugf("Output accepted batch of %d metrics in %s", len(metrics), elapsed)
```

Comment:
The original issue is that this "wrote" message shows up even after an error. Yet this is supposed to print only when there is no error? I haven't looked but should something change in the influxdb output to return an error to avoid the message in the first place?

If that isn't feasible, would it be possible to move this debug message up to always print and say "[processing|sending] N metrics to output"?

Reply:
The problem is not so much the order of the messages, but that they contradict each other. This is a case where the data hasn't been written to the DB due to a server error. The output plugin knows there's no use retrying. It seems natural for the plugin to report an error to the agent code but if it does, the agent will retry. Instead it returns nil error so the agent won't retry.

This arrangement works until you turn on debugging logging. Then the plugin says "error writing batch" followed immediately by the agent saying "wrote batch". That's confusing to users reading the debug logs.

I thought we could change "wrote batch" to something more general and say the output just "accepted" a batch and not imply that anything successful happened to the batch. Then the debug messages would reflect more closely what the code actually does.

If the output plugin API wasn't used by many other plugins, I would look into changing the signature of Write() so it was able to report more than two states. Currently Write() can return "problem/retry" or "no problem/no retry" but there ought to be a third: "problem/no retry".

I guess we could have Write() return a specific kind of error object (call it "nonRetryError") and have the agent check the error type. Then the agent could output more specific debug messages. Would you prefer doing that?

Reply:
Returning an error seems like the right thing to do.

Reply:
@powersj we cannot return an error as in this case, Telegraf will try to resend the batch of metrics while the output decided to simply drop the messages. This is mostly the case if you e.g. encounter a serialization error or similar and _know_ that there is no way this will ever succeed.